core/src/main/scala/cats/data/NonEmptyMapImpl.scala

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2018 Luka Jacobowitz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats
+package data
+
+import cats.instances.sortedMap._
+import cats.kernel._
+import cats.{Always, Apply, Eval, Foldable, Functor, Later, NonEmptyTraverse, Now, SemigroupK, Show}
+
+import scala.collection.immutable._
+
+private[data] object NonEmptyMapImpl extends NonEmptyMapInstances with Newtype2 {
+
+  private[cats] def create[K, A](m: SortedMap[K, A]): Type[K, A] =
+    m.asInstanceOf[Type[K, A]]
+
+  private[cats] def unwrap[K, A](m: Type[K, A]): SortedMap[K, A] =
+    m.asInstanceOf[SortedMap[K, A]]
+
+  def fromMap[K: Order, A](as: SortedMap[K, A]): Option[NonEmptyMap[K, A]] =
+    if (as.nonEmpty) Option(create(as)) else None
+
+  def fromMapUnsafe[K: Order, A](m: SortedMap[K, A]): NonEmptyMap[K, A] =
+    if (m.nonEmpty) create(m)
+    else throw new IllegalArgumentException("Cannot create NonEmptyMap from empty map")
+
+  def apply[K: Order, A](head: (K, A), tail: SortedMap[K, A]): NonEmptyMap[K, A] =
+    create(SortedMap(head)(Order[K].toOrdering) ++ tail)
+
+
+  def of[K: Order, A](a: (K, A), as: (K, A)*): NonEmptyMap[K, A] =
+    create(SortedMap(as: _*)(Order[K].toOrdering) + a)
+
+  def one[K: Order, A](k: K, a: A): NonEmptyMap[K, A] =
+    create(SortedMap((k, a))(Order[K].toOrdering))
+
+  implicit def catsNonEmptyMapOps[K, A](value: Type[K, A]): NonEmptyMapOps[K, A] =
+    new NonEmptyMapOps(value)
+
+
+}
+
+sealed class NonEmptyMapOps[K, A](val value: NonEmptyMap[K, A]) {
```

Comment:
Maybe `private val value`?

Reply:
These are just the Ops, what do we gain from making the `value` public? :)

Reply:
Nothing, that's why I suggested making it private :-)
In fact, I think it shouldn't be a `val` at all.

Reply:
Oh yeah, I got it switched around in my head :smile: 
Currently all of our Ops Syntax classes look this way and consistency is good the way I see it. 
What would it be if not a `val`?

Reply:
Adding `val` makes it a member on the class, omitting it, it's only in the constructor.

Reply:
Huh, yeah that makes sense, I wonder why that's the convention for syntax classes in cats, maybe someone else can chime in 🤔

Reply:
@LukaJCB you have to make it public (ie `val`) if it's a value class, which is why you see it on other syntax classes.

Reply:
If I remember correctly the reason it must be public is 2.10. For 2.11 and 2.12 `private` would work as well.